### ISSUE TITLE
Custom components in Store

### ISSUE THREAD

**Issue by kid-three** [vtex-apps/store-discussion#333]
**What are you trying to accomplish? Please describe.**  
I am trying to wrap my head around how one goes about creating custom components for a store. Do I need to fill out [the application form ](https://vtex.io/docs/recipes/development/filling-the-application-form-for-development/) for any custom app I create? 

For example, I need to create a custom Header with functionality that the default header cannot provide.  I know how to create it as its own block, but do I need to submit an application for such a simple "app'??

**What have you tried so far?**  
Taken both classes in [VTEX training course](https://lab.github.com/vtex-trainings) so I understand how the store framework works and how blocks work. I just dont understand the process to make it public

| Account | Workspace |
thenorthfacemx | kinich2

**Comment by gris** [comment]
Hello @kid-three!

I'm assuming that you want to publish your app using the same vendor name of the account you want to install right? Which would be `thenorthfacemx`.

If your app uses `react` or `node` builder you need to fill the form. These 2 are in closed beta for now and only selected customers have access to them. 

Does that answer your question?

**Comment by klzns** [comment]
Can you provide a print of the header you are trying to accomplish? Maybe you can achieve it with the native blocks

**Comment by kid-three** [comment]
Can an app include the store theme as well?

Because I need a few custom things. For example:

- An image viewer with zoom.
- Minicart where you can change the quantity of the products in the cart.
- A newsletter that has a checkbox underneath where you agree to certain terms and conditions
- A sticky header that changes layout when you scroll.
- A sticky 'call to action' on the product page so "add to cart" sticks underneath the header when scrolling
- etc..

And as I see it now I have to create an app for each one of those??

I've seen an example where react components where in the store theme and it is counted as one App.
![image](https://user-images.githubusercontent.com/32473931/86633679-93916380-bf96-11ea-8c13-9fab017ae422.png)

If I try something similar with my account I get the following error when trying to link it

```
error: Message: Account thenorthfacemx is not allowed for developing app "north-face-store" in major 0.
You can request access by going to https://forms.gle/f7bYdTA7tfdfB5tt7.
```
But I can link simple apps with react, like the `STORE-BLOCK` from VTEX TRAININGS.

I am just trying to understand how one goes about developing a store with a lot of custom components. 
Does every single little custom component have to go through the approval process?  Or can I make one app that encompases all those things in one?

**Comment by gris** [comment]
We strongly advise separating each app by functionality as described in this [document](https://vtex.io/docs/recipes/development/filling-the-application-form-for-development/). It is technically possible to bundle all this functionality in a single big app.

But in the end, deploys will become more dangerous, error tracking will be hindered too because you will not know which parts have a problem if the store is broken. Another point that can be a pain is that the development experience gets worse too. Since you have to update a bigger folder to the cloud to start developing.

In summary, create each functionality as an app separately and make your theme use them as a dependency when needed.

**Comment by kid-three** [comment]
Thank you @gris . You've been very helpful.

Just to be clear, for what I listed, I would have to make an individual app, then submit a form for each and every one of them?

Thanks,

**Comment by gris** [comment]
Correct. You have to submit one form per app 😁

**Comment by klzns** [comment]
> An image viewer with zoom.

That is possible using this feature:
https://github.com/vtex-apps/store-components/pull/794

> Minicart where you can change the quantity of the products in the cart.

That is already possible too. Using minicart.v2 default implementation
![image](https://user-images.githubusercontent.com/284515/86859319-c1f76780-c098-11ea-99cf-43d64dfe8393.png)
